### PR TITLE
[tests-only][full-ci]Add GUI test related to Sync zip file

### PR DESCRIPTION
**Pull request by Salipa-Gurung**
### Description 
This pr adds test cases related to ``` Sync works for .zip/.rar files with elaborate internal folder structures ```

### Related Issue
https://github.com/owncloud/QA/issues/789

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-ocis</strong> https://drone.owncloud.com/owncloud/client/15264/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/15264/ocis/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/15264/ocis/guiReportUpload/serverlog.log

Screenshots:
- [Syncing_a_zip_file_to_the_server.png](https://cache.owncloud.com/public/owncloud/client/15264/ocis/guiReportUpload/screenshots/Syncing_a_zip_file_to_the_server.png)
- [Syncing_folders_each_having_500_files.png](https://cache.owncloud.com/public/owncloud/client/15264/ocis/guiReportUpload/screenshots/Syncing_folders_each_having_500_files.png)

</details>

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/15323/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/15323/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/15323/oc10/guiReportUpload/serverlog.log

Screenshots:
- [Syncing_folders_each_having_500_files.png](https://cache.owncloud.com/public/owncloud/client/15323/oc10/guiReportUpload/screenshots/Syncing_folders_each_having_500_files.png)

</details>

File: test/gui/shared/steps/file_context.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import re
 import builtins
 import shutil
+import zipfile
+from zipfile import ZipFile
```
Comment (saw-jan) [inline test/gui/shared/steps/file_context.py:8]:
```suggestion
import zipfile
```

File: test/gui/tst_syncZipFile/test.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Feature: .zip files sync
+
+    As a user
+    I want to be able to upload zip file into the sync root
+    So that extracted files and folders are available on the server
+
+
+    Scenario: Syncing a zip file to the server
+        Given user "Alice" has been created on the server with default attributes and without skeleton files
+        Given the user has created a zip file "archive.zip" with following resources in temp folder
```
Comment (saw-jan) [inline test/gui/tst_syncZipFile/test.feature:10]:
```suggestion
        Given the user has created a zip file "archive.zip" with the following resources in the temp folder
```

File: test/gui/tst_syncZipFile/test.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Feature: .zip files sync
+
+    As a user
+    I want to be able to upload zip file into the sync root
+    So that extracted files and folders are available on the server
```
Comment (saw-jan) [inline test/gui/tst_syncZipFile/test.feature:5]: I would suggest to not add new folder for this one scenario. Maybe you can add it to already existing feature file

File: test/gui/tst_syncing/test.feature [hunk 4]
```diff
@@ -366,3 +366,22 @@ Feature: Syncing files
         And the user cancels the sync connection wizard
         Then the account with displayname "Alice Hansen" and host "%local_server_hostname%" should be displayed
         And the sync folder should not be added
+
+
+    Scenario: Syncing a zip file to the server
+        Given the user has created a zip file "archive.zip" with following resources in the temp folder
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:372]:
```suggestion
        Given the user has created a zip file "archive.zip" with the following resources in the temp folder
```

File: test/gui/tst_syncing/test.feature [hunk 5]
```diff
@@ -366,3 +366,22 @@ Feature: Syncing files
         And the user cancels the sync connection wizard
         Then the account with displayname "Alice Hansen" and host "%local_server_hostname%" should be displayed
         And the sync folder should not be added
+
+
+    Scenario: Syncing a zip file to the server
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:371]:
```suggestion
    Scenario: extract a zip file in the sync folder
```

File: test/gui/tst_syncing/test.feature [hunk 6]
```diff
@@ -366,3 +366,22 @@ Feature: Syncing files
...
+        And user "Alice" has set up a client with default settings
+        When user "Alice" moves file "archive.zip" from the temp folder into the sync folder
+        And user "Alice" unzips the zip file "archive.zip" inside the sync root
+        And the user waits for the files to sync
+        Then as "Alice" folder "folder1" should exist on the server
+        And as "Alice" folder "folder2" should exist on the server
+        And as "Alice" file "file1.txt" should exist on the server
+        And as "Alice" file "file2.txt" should exist on the server
+        And the content of file "file1.txt" for user "Alice" should be "Test file1" on the server
+        And the content of file "file2.txt" for user "Alice" should be "Test file2" on the server
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:387]: should be enough
```suggestion
        And the content of file "file1.txt" for user "Alice" should be "Test file1" on the server
        And the content of file "file2.txt" for user "Alice" should be "Test file2" on the server
```

File: test/gui/tst_syncing/test.feature [hunk 7]
```diff
@@ -366,3 +366,22 @@ Feature: Syncing files
...
         And the sync folder should not be added
+
+
+    Scenario: extract a zip file in the sync folder
+        Given the user has created a zip file "archive.zip" with the following resources in the temp folder
+            |resource  |type   |content    |
+            |folder1   |folder |           |
+            |folder2   |folder |           |
+            |file1.txt |file   |Test file1 |
+            |file2.txt |file   |Test file2 |
```
Comment (saw-jan) [inline test/gui/tst_syncing/test.feature:377]:
```suggestion
            | resource  | type   | content    |
            | folder1   | folder |            |
            | folder2   | folder |            |
            | file1.txt | file   | Test file1 |
            | file2.txt | file   | Test file2 |
```

**Review by saw-jan — COMMENTED** [review]
small changes

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**